### PR TITLE
feat(quick-search): Add title prop to QuickSearch for SelectorDropdown

### LINKED ISSUE #2857: Allow QuickSearch to pass "title" to SelectorDropdown
**Describe the solution you'd like**
Need the ability to pass a node to the SelectorDropdown component used in the QuickSearch that is rendered outside of the "overlay list". The SelectorDropdown already accepts the optional "title" prop and I would like the ability for the QuickSearch to accept the same thing and pass it down.

**Describe alternatives you've considered**
In order to render anything in the overlay-wrapper and outside of the overlay list, I would need to create my own QuickSearch component.

**Additional context**
I am a Box employee

### PR DESCRIPTION
**Pull request by austindbertola**
fixes #2857

### CODE REVIEW

File: src/features/quick-search/__tests__/QuickSearch.test.js [hunk 1]
```diff
@@ -107,6 +107,23 @@ describe('features/quick-search/QuickSearch', () => {
...
+        const title = <div>title</div>;
+        const wrapper = shallow(
+            <QuickSearch title={title} inputProps={inputProps}>
+                {children}
+            </QuickSearch>,
+        );
+        const selectorDropdown = wrapper.find('SelectorDropdown');
+        expect(selectorDropdown.prop('title')).toEqual(title);
+    });
+
```
Comment (tjuanitas) [inline src/features/quick-search/__tests__/QuickSearch.test.js:126]: I think this is meant to be inside the `describe('render()', () => {...}` block?

Comment (austindbertola) [reply src/features/quick-search/__tests__/QuickSearch.test.js]: ah good catch. I moved it.

**Comment by tjuanitas** [comment]
Can you add before and after screenshots of the component with the new prop?

**Comment by austindbertola** [comment]
@tjuanitas I created a comment in the linked issue with screenshots. Let me know if that is sufficient: https://github.com/box/box-ui-elements/issues/2857#issuecomment-1071945323

**Review by tjuanitas — APPROVED** [review]

**Pull request merged**